app/scripts/controllers/permissions/caveat-mutators.js

```diff
@@ -10,6 +10,7 @@ export const CaveatMutatorFactories = {
   [CaveatTypes.restrictReturnedAccounts]: {
     removeAccount,
   },
+  [CaveatTypes.restrictNetworkSwitching]: {},
```

Comment:
This is only necessary if we call `updatePermissionsByCaveat` for the given caveat, which we do not. However, if a network is completely removed from the user's wallet, we _should_ make use of this. See `removeAllAccountPermissions` in `metamask-controller.js` for prior art.